ddtelemetry/src/config.rs

```diff
@@ -51,40 +52,50 @@ fn endpoint_with_telemetry_path(
 /// Settings gathers configuration options we receive from the environment
 /// (either through env variable, or that could be set from the )
 pub struct Settings {
-    pub agent_host: String,
-    pub trace_agent_port: u16,
-    pub trace_agent_url: Option<Uri>,
+    // Env parameter
+    pub agent_host: Option<String>,
+    pub trace_agent_port: Option<u16>,
+    pub trace_agent_url: Option<String>,
+    pub trace_pipe_name: Option<String>,
```

Comment:
I assume only some of these can be set at once? Would it be possible to use an `Enum` to encode that?

Reply:
This struct is a wrapper around env variables for easier testing (you can look at this file's tests for instance).
All of them can be set at once, but some of them take precedence over each other ([see here](https://github.com/DataDog/architecture/blob/ee430517f3435cd686a7236e4b1b0a9715ef8b7c/rfcs/apm/integrations/trace-autodetect-agent-config/rfc.md#explicit-configuration)), and the logic on how is in Configuration::from_setting.